backend/eventloop/shared_resource_loop/sharedresourceloop_repositorycredential_test.go

```diff
@@ -0,0 +1,220 @@
+				normalizedRepoUrl string
+			}{
+				{
+					repoUrl: "https://github.com/redhat-appstudio/test.git", normalizedRepoUrl: "https://github.com/redhat-appstudio/test",
+				},
+			}
+
+			for _, data := range testData {
+				Expect(NormalizeGitURL(data.repoUrl)).To(Equal(data.normalizedRepoUrl))
+			}
+		})
+	})
+
+	Context("Set GitOpsDeploymentRepositoryCredentials status conditions", func() {
+
+		var (
+			ctx                                    context.Context
+			k8sClient                              client.Client
+			gitopsDeploymentRepositoryCredentialCR *managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredential
+		)
+
+		BeforeEach(func() {
+			scheme, _, _, workspace, err := tests.GenericTestSetup()
+			Expect(err).To(BeNil())
+
+			ctx = context.Background()
+
+			gitopsDeploymentRepositoryCredentialCR = &managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredential{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-repocred",
+					Namespace: workspace.Name,
+				},
+			}
+
+			k8sClient = fake.NewClientBuilder().WithScheme(scheme).WithObjects(gitopsDeploymentRepositoryCredentialCR).Build()
+
+		})
+
+		AfterEach(func() {
+			err := k8sClient.Delete(ctx, gitopsDeploymentRepositoryCredentialCR)
+			Expect(err).To(BeNil())
+		})
+
+		var haveErrOccurredConditionSet = func(expectedRepoCredStatus managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialStatus) matcher.GomegaMatcher {
+
+			// sanitizeCondition removes ephemeral fields from the GitOpsDeploymentCondition which should not be compared using
+			// reflect.DeepEqual
+			sanitizeCondition := func(cond *metav1.Condition) metav1.Condition {
+
+				res := metav1.Condition{
+					Type:   cond.Type,
+					Status: cond.Status,
+					Reason: cond.Reason,
+				}
+
+				return res
+
+			}
+			return WithTransform(func(repoCred *managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredential) bool {
+
+				if err := k8sClient.Get(ctx, client.ObjectKeyFromObject(repoCred), repoCred); err != nil {
+					GinkgoWriter.Println(err)
+					return false
+				}
+
+				conditionExists := false
+				existingConditionList := repoCred.Status.Conditions
+
+				if len(expectedRepoCredStatus.Conditions) != len(existingConditionList) {
+					fmt.Println("HaveConditions:", conditionExists, "/ Expected:", expectedRepoCredStatus.Conditions, "/ Actual:", repoCred.Status.Conditions)
+					return false
+				}
+
+				for _, resourceCondition := range expectedRepoCredStatus.Conditions {
+					conditionExists = false
+					for _, existingCondition := range existingConditionList {
+						if reflect.DeepEqual(sanitizeCondition(resourceCondition), sanitizeCondition(existingCondition)) {
+							conditionExists = true
+							break
+						}
+					}
+					if !conditionExists {
+						fmt.Println("GitOpsDeploymentCondition:", conditionExists, "/ Expected:", expectedRepoCredStatus.Conditions, "/ Actual:", repoCred.Status.Conditions)
+						break
+					}
+				}
+				return conditionExists
+
+			}, BeTrue())
+		}
+
```

Comment:
We should also test the actual `UpdateGitopsDeploymentRepositoryCredentialStatus` functionality itself, for example, creating a `GitOpsDeploymentRepositoryCredential` with a valid/invalid URL, and making sure the conditions are set properly.

Reply:
I have added this as part of e2e tests.

Reply:
That's true: I think we can just query Git (e.g. GitHub) directly, as part of the test, rather than mocking it. If that causes us problems later (for example, due to rate limiting), we can re-examine again.

Reply:
The problem with using Git tests as part of unit tests is that we would need to store username/password as well for testing the success scenario. I have added tests for wrong secrets, but I don't think we can test success scenario as we would need to store real username and password for the test.